src/webgpu/api/operation/buffers/map_oom.spec.ts

```diff
@@ -45,15 +46,32 @@ g.test('mapAsync')
         }),
       oom
     );
-    const promise = t.expectGPUError(
-      'validation', // Should be a validation error since the buffer is invalid.
-      () => buffer.mapAsync(write ? GPUMapMode.WRITE : GPUMapMode.READ),
-      oom
-    );
+
+    let promise: Promise<void>;
+    // Should be a validation error since the buffer is invalid.
+    // Unmap abort error shouldn't cause a validation error.
+    t.expectValidationError(() => {
```

Comment:
Maybe we could make `expectValidationError` generic on the return type of `fn` like `expectGPUError` so that we can do:

```
const promise = t.expectValidationError(()=> buffer.mapAsync(,,,));
```

and we don't need the addition of `!` below.

Reply:
I've considered this but avoided it because it breaks a sanity check we have right now that the expectValidationError callback doesn't return a Promise. This is because if you write an async function there it _looks_ like the scope is around the whole thing, but it's only up to the first await.

However I did have an idea when reviewing this PR the other day, we could change it to check whether the function itself is async instead of checking the return value. People online say not to do this because async function often get transpiled into non-async functions, but in our case our build steps never do that.

In any case it's be complicated enough I think it should be a followup.

Reply:
Thanks for the suggesion. Let's make another PR for improvement.